lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/LeastSquaresExpansion.cxx

```diff
@@ -0,0 +1,195 @@
+    const Sample & outputSample,
+    const Distribution & distribution)
+  : FunctionalChaosAlgorithm(inputSample, weights, outputSample, distribution)
+{
+  // Extract the basis from the adaptive strategy attribute of the base class
+  basis_ = getAdaptiveStrategy().getBasis();
+  // The basis size is the maximum dimension of the adaptive strategy
+  basisSize_ = getAdaptiveStrategy().getMaximumDimension();
+  // The method name is set to the default one, given by ResourceMap
+  methodName_ = ResourceMap::GetAsUnsignedInteger("FunctionalChaosAlgorithm-LeastSquaresMethod");
+}
+
+/* Constructor */
+LeastSquaresExpansion::LeastSquaresExpansion(const Sample & inputSample,
+    const Sample & outputSample,
+    const Distribution & distribution,
+    const OrthogonalBasis & basis,
+    const UnsignedInteger basisSize,
+    const String & methodName)
+  : LeastSquaresExpansion(inputSample, Point(inputSample.getSize(), 1.0), outputSample, distribution, basis, basisSize, methodName)
+{
+  // Nothing to do  
+}
+
+/* Constructor */
+LeastSquaresExpansion::LeastSquaresExpansion(const Sample & inputSample,
+    const Point & weights,
+    const Sample & outputSample,
+    const Distribution & distribution,
+    const OrthogonalBasis & basis,
+    const UnsignedInteger basisSize,
+    const String & methodName)
+  : FunctionalChaosAlgorithm(inputSample, weights, outputSample, distribution)
+  , basis_(basis)
+  , basisSize_(basisSize)
+  , methodName_(methodName)
+{
+  // The arguments are checked in the base class, excepted the basis and its size
+  if (basis.getMeasure().getDimension() != distribution.getDimension()) throw InvalidArgumentException(HERE) << "Error: the basis must have a measure with the same dimension as the input distribution, heare measure dimension=" << basis.getMeasure().getDimension() << " and distribution dimension=" << distribution.getDimension();
+  if (basisSize == 0) throw InvalidArgumentException(HERE) << "Error: cannot project on a basis of size zero";
+}
+
+
+/* Virtual constructor */
+LeastSquaresExpansion * LeastSquaresExpansion::clone() const
+{
+  return new LeastSquaresExpansion(*this);
+}
+
+
+/* String converter */
+String LeastSquaresExpansion::__repr__() const
+{
+  return OSS() << "class=" << getClassName();
+}
+
+
+/* Computes the functional chaos */
+void LeastSquaresExpansion::run()
+{
+  // Check if we have to transform the input sample in order to adapt it
+  // to the basis
+  const Distribution measure(basis_.getMeasure());
+  const Bool noTransformation = (distribution_ == measure);
+  Sample transformedInputSample;
+  DistributionTransformation transformation;
+  if (noTransformation) transformedInputSample = inputSample_;
+  else
+    {
+      transformation = DistributionTransformation(distribution_, basis_.getMeasure());
+      transformedInputSample = transformation(inputSample_);
+    }
+  // Build the relevant functions from the basis
+  Indices indices(basisSize_);
+  indices.fill();
```

Comment:
I suggest to provide a new method so that we can set the indices. There are two reasons for this. Both require to first compute the indices from model selection, then use it to : 
- compute analytical LOO Q2, 
- perform bootstrap PCE.

This is a very small change in the interface, but provides a feature that is not available in the current implementation.

Reply:
@mbaudin47 This point has been discussed during the dedicated meeting last Monday, and it was decided to postpone this feature as it required significant additional work to make it fast. As the RC has been postponed too, I will do my best to add the feature, but through a method and not the constructor in order to benefit from a reusable internal state.

Reply:
@regislebrun : I understand now the point, which I did not do at meeting time. You do not want to directly set the list of indices in the specific 2 cases that I mentioned, because the basis functions would be re-created again and again: building the basis will limit the speed. This is because I do not to create a FCExpansion object one single time, but repeat that evaluation several times with different input samples, but with a constant basis. I now understand that the code that I am currently using, which directly uses the `DesignProxy`, is more efficient than a naive approach: I did not think of that point when I used it.

@sofianehaddad : Defining an `EnumerateFunction` based on a list of integers instead of directly using the `Indices` sounds overkill to me.

To clarify the topic, I will write a use-case below soon.

Reply:
@mbaudin47 I don't think. We set a list of tools including `EnumerateFunction` to implement the `PCE` algorithm. Whatever the sraegy you have in mind, you need an **orthogonal basis** which is built upon an infinity of functions (tensorized polynoms) and a truncature (+a list of indices "to explore")., defined through an enumerate function. 

At the end of the algorithm, you **get a finite list of indices**. IMHO such a result object should not be used like that directly in `PCE`

Other thing, imagine you have your enumerate function with the list of indices, then within the `LARS` you will be able to get sequentially the LOO/K-Fold errors

Thus definitively I think we should benefit from this new chaos implementation to maintain some rigor!

If you still have in mind to use the **list directly**,  we need to rethink ctors...

Reply:
Hi all, I hope I found a way to allow for a direct selection of indices while preserving the constructors and allowing for efficient computations. It needed to make the DesignProxy class persistent and a few other changes, and it is not ready for publication yet.
I changed the way the expansions are tested, comparing the computed coefficients to the reference ones as it is the main purpose of the algorithm and not the Sobol indices. BTW I found something strange with the automatic construction of a basis when the weights are not uniform, still under investigation!

It should be ok by the end of the day, despite the sun here in Britany (here, Charlotte on the top of the local mountain)! 
![IMG_2612](https://user-images.githubusercontent.com/8220933/235612045-e919f2a0-7731-4f19-939a-dec65649a341.JPG)

Reply:
What if I use a `Basis` instead of an `OrthogonalBasis`? This way, I could set the basis once for all and use all the indices that are computed by `indices.fill()` using the current code. This basis could be set as a collection of functions which correspond to a selection computed from a previous run. After all, computing the coefficients from an orthogonal basis or a general basis does not change much of the code. The only difference between `Basis` and `OrthogonalBasis` is that part of the code where we call the `build()` method which must be removed if the `Basis` is given instead of `OrthogonalBasis`. In other words:
- if an `OrthogonalBasis` is given, this means that we must create the collection of functions using `build()` from an arbitrary size collection of functions,
- if a `Basis` is given, this means that a constant collection of candidate functions is used. No `build()` is required here.

The potential lack of orthogonality _does_, however, change the interpretation of the `FCResult`, but this could be restricted afterwards.

Reply:
All the functional chaos related classes assume the orthonormality of the basis:
 * It is the only setting in which the components of the orthogonal projection of the function of interest on the vector space spanned by the basis can be computed either by minimizing the norm of the residual (the least-squares approach) and by computing the dot-product wrt the bases functions.
 * It is the only setting in which the result can be used to compute Sobol indices the way we do it.
 * In the current implementation we check if the given distribution (i.e. the given weighted input sample) is adapted to the given basis, and we introduce the relevant transformation if needed. What do you suggest to do if the given basis doesn't have an associated measure?

So the change is not only in the interpretation of the result but also in the actual computation of the projection.

I can see a way to adapt it to this more general setting, but will not do it in this PR due to a lack of time (I have other open PRs to be merged in the coming release).